lib/web_ui/lib/src/engine.dart

```diff
@@ -22,10 +24,68 @@ import 'package:meta/meta.dart';
 
 import '../ui.dart' as ui;
 
-part 'engine/alarm_clock.dart';
+import 'engine/alarm_clock.dart';
+export 'engine/alarm_clock.dart';
+
+import 'engine/browser_detection.dart';
+export 'engine/browser_detection.dart';
+
+import 'engine/mouse_cursor.dart';
+export 'engine/mouse_cursor.dart';
+
+import 'engine/navigation/history.dart';
+export 'engine/navigation/history.dart';
+
+import 'engine/navigation/js_url_strategy.dart';
+export 'engine/navigation/js_url_strategy.dart';
+
+import 'engine/navigation/url_strategy.dart';
+export 'engine/navigation/url_strategy.dart';
```

Comment:
Something that might make this files slightly more compact (but still maintaining modularity), is to make a `engine/navigation/navigation.dart` that exports everything it needs to export from within the `navigation` package.

That way, in this more global package, you just need to do: `export engine/navigation/navigation.dart`.

If you prefer to have more control, you can then `show` only what's really exported as part of the public API of the engine, vs what was available to engine programmers.

We've done this here, for example:
* https://github.com/flutter/plugins/blob/master/packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/google_maps_flutter_platform_interface.dart (see types/types.dart)

And then, from the google maps, only reexport (via a show) what we explicitly want end users to see from the package above:
* https://github.com/flutter/plugins/blob/master/packages/google_maps_flutter/google_maps_flutter/lib/google_maps_flutter.dart

(Note that the last link still has two part'ed files, because there's a cross-dependency between them! :P)

Reply:
Makes sense! I'll do this.